packages/@lwc/synthetic-shadow/src/shared/node-collection-util.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isUndefined, isTrue } from './language';
+/**
+ * Writing our own utils to handle NodeList and HTMLCollection. This is to not conflict with
+ * some legacy third party libraries like prototype.js that patch Array.prototype.
+ */
+
+/**
+ * Custom implementation of filter since using Array.prototype.filter conflicts with other
+ * legacy libraries like prototype.js
+ * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Polyfill
+ */
+export function collectionFilter<T extends Node, K extends Element>(
+    this: NodeListOf<T> | HTMLCollectionOf<K>,
+    fn: (value: T | K, index?: number, collection?: NodeListOf<T> | HTMLCollectionOf<K>) => boolean,
+    thisArg?: any
+): Array<T | K> {
+    const res: Array<T | K> = [];
+    const length = this.length;
+    for (let i = 0; i < length; i++) {
+        const curr = this[i];
+        if (isTrue(fn.call(thisArg, curr, i, this))) {
+            res.push(curr);
```

Comment:
should this use ArrayPush instead?

Reply:
I keep forgetting to use the shared method. 🤦‍♂